firewood/src/merkle/node.rs

```diff
@@ -402,6 +416,113 @@ impl<T> PartialEq for EncodedNode<T> {
     }
 }
 
+#[derive(Debug)]
+struct PathWithBitsPrefix(Path);
```

Comment:
Why not 
```rust
struct PathWithBitsPrefix<T>(T, Path);
```
?
(I just use `T`, because I don't know the actual size, could be a `u64` or a `[u8; N]`)